### PR TITLE
[CE Sprint #16] [CLOUD-2398] Introduce ephemeral HTTPS and MySQL/PostgreSQL persistent templates utilizing OpenShift's serving x509 certificate secrets service

### PR DESCRIPTION
**Pull request by iankko**
<br/>
[CLOUD-2398] Add ephemeral HTTPS template, and two persistent templates (MySQL one, PostgreSQL one) to deploy RH-SSO server with the auto-generated HTTPS and JGroups keystores, and an auto-generated truststore for the RH-SSO server
<br/><br/>
While introducing these templates, also correct the ```HOSTNAME_HTTP```/```HOSTNAME_HTTPS``` CLOUD-2367 issue in them (do not introduce it again)
<br/><br/>
Signed-off-by: Jan Lieskovsky <jlieskov@redhat.com>
<br/><br/>
Testing status: All three of the proposed templates have been tested on local OCP v3.7 install, and all of them seem to be working correctly AFAICT.
<br/>

### CODE REVIEW

File: sso/sso72-x509-https.json [hunk 1]
```diff
@@ -0,0 +1,521 @@
...
+            "displayName": "Application Name",
+            "description": "The name for the application.",
+            "name": "APPLICATION_NAME",
+            "value": "sso",
+            "required": true
+        },
+        {
+            "displayName": "Server HTTPS Keystore Password",
+            "description": "The password for the HTTPS keystore.",
+            "name": "HTTPS_PASSWORD",
```
Comment (maschmid) [inline sso/sso72-x509-https.json]: why is this needed in this template?

Comment (iankko) [reply sso/sso72-x509-https.json]: Just this way seemed more elegant to me, than to do something like:
```
HTTPS_PASSWORD=$(openssl rand -base64 32)
```
or
```
PASSWORD=$(openssl rand -base64 32)
HTTPS_PASSWORD=${PASSWORD}
```
directly in the ```openshift-x509.sh``` script (like the user having the ability to see the password, seemed more transparent way to me).

But if it's not needed (the user doesn't need to be aware of the various passwords), I can use the ```openssl rand``` approach above (remove all three of them).

**Comment by iankko** [comment]
The templates have been reduced and simplified per feedback on CLOUD-2398. Also removed the unnecessary third volume from them.

File: sso/sso72-x509-https.json [hunk 2]
```diff
@@ -0,0 +1,415 @@
...
+                                        "name": "OPENSHIFT_DNS_PING_SERVICE_NAME",
+                                        "value": "${APPLICATION_NAME}-ping"
+                                    },
+                                    {
+                                        "name": "OPENSHIFT_DNS_PING_SERVICE_PORT",
+                                        "value": "8888"
+                                    },
+                                    {
+                                        "name": "X509_CRT_DELIMITER",
+                                        "value": "/-----BEGIN CERTIFICATE-----/"
```
Comment (maschmid) [inline sso/sso72-x509-https.json]: should we perhaps default this in the image so that the template doesn't need to specify it?

Comment (iankko) [reply sso/sso72-x509-https.json]: Ok, will update yet.
